e2e/test/scenarios/embedding/embedding-smoketests.cy.spec.js

```diff
@@ -74,7 +74,7 @@ describe("scenarios > embedding > smoke tests", { tags: "@OSS" }, () => {
         "With the embedding enabled, we should now see two new sections on the main page",
       );
       cy.log("The first section: 'Static embedding'");
-      cy.findByTestId("-static-embedding-setting").within(() => {
+      cy.findByRole("article", { name: "Static embedding" }).within(() => {
```

Comment:
We stop relying on settings magic. [And opted for `component` API for admin settings instead](https://www.notion.so/Tech-Feature-flag-and-settings-for-SDK-9945e76d7e80404fa79b7d5b93049978?d=c507d778272445ab886b722db44cea1d&pvs=4) as suggested by admin webapp team.

I fix this by making this UI more accessible and use typical selectors insead.
https://github.com/metabase/metabase/blob/191d832d8fac25f1c0f510d0ad92b12cc2e460e6/frontend/src/metabase/admin/settings/components/widgets/EmbeddingOption/EmbeddingOption.tsx#L41

Reply:
I like this approach!